src/server/src/main/java/io/cassandrareaper/jmx/ClusterFacade.java

```diff
@@ -439,45 +439,49 @@ public String getLocalEndpoint(Node node) throws ReaperException, InterruptedExc
    * List running compactions on a specific node either through JMX or through the backend.
    *
    * @param node the node to get the compactions from.
-   * @return a list of compactions
+   * @return a number of pending compactions and a list of compactions
    * @throws MalformedObjectNameException ¯\_(ツ)_/¯
    * @throws ReflectionException ¯\_(ツ)_/¯
    * @throws ReaperException any runtime exception we catch in the process
    * @throws InterruptedException in case the JMX connection gets interrupted
    * @throws IOException errors in parsing JSON encoded compaction objects
    */
-  public List<Compaction> listActiveCompactions(Node node)
+  public CompactionStats listActiveCompactions(Node node)
       throws MalformedObjectNameException, ReflectionException, ReaperException, InterruptedException {
 
     String nodeDc = getDatacenter(node);
     if (nodeIsAccessibleThroughJmx(nodeDc, node.getHostname())) {
       // We have direct JMX access to the node
-      return listActiveCompactionsDirect(node);
+      return listCompactionStatsDirect(node);
     } else {
       // We don't have access to the node through JMX, so we'll get data from the database
       LOG.info("Node {} in DC {} is not accessible through JMX", node.getHostname(), nodeDc);
 
       String compactionsJson = ((IDistributedStorage)context.storage)
           .listOperations(node.getClusterName(), OpType.OP_COMPACTION, node.getHostname());
 
-      return parseJson(compactionsJson, new TypeReference<List<Compaction>>(){});
+      return parseJson(compactionsJson, new TypeReference<CompactionStats>(){});
```

Comment:
That could blow up during an upgrade. It's not backwards compatible with the previous storage format, so data in the database could come with the `Compaction` type instead of `CompactionStats`.
We'd need to gracefully ignore that, right?

Reply:
Yes, that's a good catch. I've just pushed a commit fixing this.